cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/batch/sources/StreamBatchSource.java

```diff
@@ -60,73 +68,59 @@
-      "'format.setting.' will be passed to the format. For example, if a property with key " +
-      "'format.setting.delimiter' and value '|' is given, the setting 'delimiter' with value '|' " +
-      "will be passed to the format.");
-    configurer.addProperty(new Property(Properties.Stream.NAME, "Name of the stream.", true));
-    configurer.addProperty(new Property(
-      Properties.Stream.DURATION,
-      "Size of the time window to read with each run of the pipeline. The format is expected to be a number followed " +
-        "by a 's', 'm', 'h', or 'd' specifying the time unit, with 's' for seconds, 'm' for minutes, 'h' for hours, " +
-        "and 'd' for days. For example, a value of '5m' means each run of the pipeline will read 5 minutes of events " +
-        "from the stream.",
-      true));
-    configurer.addProperty(new Property(
-      Properties.Stream.DELAY,
-      "Optional delay for reading stream events. The value must be of the same format as the duration value. " +
-        "For example, a duration of '5m' and a delay of '10m' means each run of the pipeline will read events from " +
-        "15 minutes before its logical start time to 10 minutes before its logical start time. The default value is 0.",
-      false));
-    configurer.addProperty(new Property(
-      Properties.Stream.FORMAT,
-      "Optional format of the stream. Any format supported by CDAP is also supported. For example, a value of 'csv' " +
-        "will attempt to parse stream events as comma separated values. If no format is given, event " +
-        "bodies will be treated as bytes, resulting in a three field schema: 'ts' of type long, " +
-        "'headers' of type map of string to string, and 'body' of type bytes.",
-      false));
-    configurer.addProperty(new Property(
-      Properties.Stream.SCHEMA,
-      "Optional schema for the body of stream events. Schema is used in conjunction with format " +
-        "to parse stream events. Some formats like the avro format require schema, while others do not. " +
-        "The schema given is for the body of the stream, so the final schema of records output by the source will " +
-        "contain an additional field named 'ts' for the timestamp and a field named 'headers' for the headers as " +
-        "as the first and second fields of the schema.",
-      false));
+  public StreamBatchSource(StreamBatchConfig streamBatchConfig) {
+    this.streamBatchConfig = streamBatchConfig;
   }
 
   @Override
   public void configurePipeline(ETLStage stageConfig, PipelineConfigurer pipelineConfigurer) {
-    Config config = new Config(stageConfig.getProperties());
-    pipelineConfigurer.addStream(new Stream(config.name));
+    pipelineConfigurer.addStream(new Stream(streamBatchConfig.name));
   }
 
   @Override
   public void prepareJob(BatchSourceContext context) {
-    config = new Config(context.getPluginProperties().getProperties());
+    config = new Config(streamBatchConfig);
```

Comment:
You shouldn't need the extra `Config` object. Just do the conversion when needed since `prepareJob` only get called once per run.